File: performance-comparison/benchmarks/numerical/python/matrix_multiply.py
import sys

def matmul(n):
    A = [[1.0] * n for _ in range(n)]
    B = [[1.0] * n for _ in range(n)]
    C = [[0.0] * n for _ in range(n)]
    for i in range(n):
        for k in range(n):
            for j in range(n):
                C[i][j] += A[i][k] * B[k][j]
    return sum(C[i][j] for i in range(n) for j in range(n))

n = int(sys.argv[1]) if len(sys.argv) > 1 else 10
print(int(matmul(n)))

File: performance-comparison/benchmarks/numerical/clojure/matrix_multiply.clj
(defn make-matrix [n init]
  (vec (repeatedly n #(vec (repeat n init)))))

(defn matmul [A B n]
  (vec
    (for [i (range n)]
      (vec
        (for [j (range n)]
          (reduce + (map #(* (get-in A [i %]) (get-in B [% j]))
                         (range n))))))))

(let [n (Integer/parseInt (first *command-line-args*))
      A (make-matrix n 1.0)
      B (make-matrix n 1.0)
      C (matmul A B n)
      s (reduce + (map #(reduce + %) C))]
  (println (long s)))
